---
title: "Instalar paquetes Bioconductor"
author: "Estefania Mancini"
date: "29 de Marzo 2022"
output:
    html_document:
    toc: true # table of content true
    toc_depth: 3  # upto three depths of headings (specified by #, ## and ###)
    number_sections: true  ## if you want number sections at each table header
    toc_float: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

## Utilizaremos los siguientes  paquetes

* [edgeR](http://bioconductor.org/packages/release/bioc/html/edgeR.html)

* [DESeq2](http://bioconductor.org/packages/release/bioc/html/DESeq2.html)

* [RNAseqData.HNRNPC.bam.chr14](http://bioconductor.org/packages/release/data/experiment/html/RNAseqData.HNRNPC.bam.chr14.html)

* [GenomicRanges](http://bioconductor.org/packages/release/bioc/html/GenomicRanges.html)

* [GenomicFeatures](http://bioconductor.org/packages/release/bioc/html/GenomicFeatures.html)

* [GenomicAlignments](http://bioconductor.org/packages/release/bioc/html/GenomicAlignments.html)

* [ggplot2](https://cran.r-project.org/)

* [pheatmap](https://cran.r-project.org/)


```{r install, echo=TRUE, eval=FALSE}

if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("RNAseqData.HNRNPC.bam.chr14")
BiocManager::install("GenomicRanges")
BiocManager::install("GenomicFeatures")
BiocManager::install("GenomicAlignments")
BiocManager::install("edgeR")
BiocManager::install("pheatmap")
BiocManager::install("DESeq2")
```

